File: C++/MergeTwoSortedLists.cpp
#include<iostream>
#include<stdlib.h>
#include<stack>
#include<string>
#include<string.h>
#include<algorithm>
#include<math.h>
#include<map>
using namespace std;

struct ListNode {
	int val;
	ListNode *next;
	ListNode(int x) : val(x), next(NULL) {}
};
class Solution {
public:
	ListNode *mergeTwoLists(ListNode *l1, ListNode *l2) {
		// Start typing your C/C++ solution below
		// DO NOT write int main() function
		if(!l1)
			return l2;
		if(!l2)
			return l1;
		ListNode* head = NULL;
		ListNode* cur = head;
		if(l1->val < l2->val)
		{
			head = cur = l1;
			l1 = l1->next;
		}
		else
		{
			head = cur = l2;
			l2 = l2->next;
		}
		while(l1 && l2)
		{
			if(l1->val <= l2->val)
			{
				cur->next = l1;
				l1 = l1->next;
			}
			else
			{
				cur->next = l2;
				l2 = l2->next;
			}
			cur = cur->next;
		}
		while(l1)
		{
			cur->next = l1;
			cur = cur->next;
			l1 = l1->next;
		}
		while(l2)
		{
			cur->next = l2;
			cur = cur->next;
			l2 = l2->next;
		}
		return head;
	}
};

File: JAVA/MergeTwoSortedLists.java
class ListNode {
	int val;
	ListNode next;

	ListNode(int x) {
		val = x;
		next = null;
	}
}

class Solution {
	public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
		// Start typing your Java solution below
		// DO NOT write main() function
		if(l1 == null && l2 == null)
			return null;
		ListNode head = null;
		if(l1 == null)
		{
			head = l2;
			l2 = l2.next;
		}
		else if(l2 == null)
		{
			head = l1;
			l1 = l1.next;
		}
		else
		{
			if(l1.val < l2.val)
			{
				head = l1;
				l1 = l1.next;
			}
			else
			{
				head = l2;
				l2 = l2.next;
			}
		}
		ListNode cur = head;
		while(l1 != null && l2 != null)
		{
			if(l1.val < l2.val)
			{
				cur.next = l1;
				l1 = l1.next;
			}
			else
			{
				cur.next = l2;
				l2 = l2.next;
			}
			cur = cur.next;
		}
		while(l1 != null)
		{
			cur.next = l1;
			l1 = l1.next;
			cur = cur.next;
		}
		while(l2 != null)
		{
			cur.next = l2;
			l2 = l2.next;
			cur = cur.next;
		}
		return head;
	}
}

public class MergeTwoSortedLists {

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		// TODO Auto-generated method stub

	}

}
